### PR TITLE
Multipart delivery (#7)

### PR DESCRIPTION
**Pull request by rosafaria**
Created delivery method with locals.

### CODE REVIEW

File: lib/lotus/mailer.rb [hunk 1]
```diff
@@ -94,5 +96,70 @@ def self.load!
...
+      #     :password             => ENV["GMAIL_PASSWORD"],
+      #     :authentication       => "plain",
+      #     :enable_starttls_auto => true
+      #   }
+      #
+      #   from ENV["GMAIL_USER"]
+      #   to "inesopcoelho@gmail.com"
+      #   subject "This is the subject"
+      #
+      #   Lotus::Mailer.configure do
```
Comment (joneslee85) [inline lib/lotus/mailer.rb]: please move it out side of this mailer

File: lib/lotus/mailer.rb [hunk 2]
```diff
@@ -94,5 +96,70 @@ def self.load!
...
+        mail['to'] = self.class.to
+        mail['subject'] = self.class.subject
+        if !(Lotus::Mailer.configuration.delivery_method.nil?)
+          mail.delivery_method *Lotus::Mailer.configuration.delivery_method
+        end
+
+        #attach templates
+        self.class.templates.each do |type,content|
+          case type
+            when :html
```
Comment (joneslee85) [inline lib/lotus/mailer.rb]: please align `when` with `case`

File: lib/lotus/mailer.rb [hunk 3]
```diff
@@ -94,5 +96,72 @@ def self.load!
...
+
+    module InstanceMethods
+      # Delivers a multipart email, by looking at all the associated templates and render them.
+      #
+      # @since 0.1.0
+      def deliver
+        mail['from'] = self.class.from
+        mail['to'] = self.class.to
+        mail['subject'] = self.class.subject
+        if !(Lotus::Mailer.configuration.delivery_method.nil?)
```
Comment (joneslee85) [inline lib/lotus/mailer.rb]: what about `if Lotus::Mailer.configuration.delivery_method`? You could check for truthiness here

File: lib/lotus/mailer.rb [hunk 4]
```diff
@@ -94,5 +96,72 @@ def self.load!
...
+          mail.delivery_method *Lotus::Mailer.configuration.delivery_method
+        end
+
+        #attach templates
+        self.class.templates.each do |type,content|
```
Comment (joneslee85) [inline lib/lotus/mailer.rb]: add a space after `,`

**Comment by joneslee85** [comment]
Please fix the spec, get the build green :) https://travis-ci.org/lotus/mailer/jobs/77319916

**Comment by rosafaria** [comment]
@joneslee85 I think it's ok now, could you check please? :)

**Pull request merged**